### PR TITLE
Features/connect slide api

### PR DESCRIPTION
**Pull request by ridwanraji**
## Description

Added  some actions and a few more mutations to be able to
- Get all slides
- Get a singe slide
- Save a new slide
- Save an edited slide (overwrite the body of slide that has already been saved)
- Delete a slide
and unit tests for the above actions

## Related Issue or Trello Card

[Trello card](https://trello.com/c/qZMWLHOl/2-connect-the-rest-apis-to-the-front-end)

## Motivation and Context

To store creation and changes made to a slide to the server

## How Has This Been Tested?

- [x] Unit test
- [ ] NightWatch test
- [ ] Coverage testing
- [x] Manual testing

## Test Cases and test paths

- Get all slides
- Get a singe slide
- Save a new slide
- Save an edited slide (overwrite the body of slide that has already been saved)
- Delete a slide

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/store/modules/slide/mutation-types.js [hunk 1]
```diff
@@ -55,3 +55,9 @@ export const SET_META_TIMEOUT = 'SET_META_TIMEOUT'
 export const SET_META_REPEATABLE = 'SET_META_REPEATABLE'
 export const SET_META_STARTDATE = 'SET_META_STARTDATE'
 export const SET_META_ENDDATE = 'SET_META_ENDDATE'
+
+// for set template
+export const SET_TEMPLATE = 'SET_TEMPLATE'
+
+// for set timeout
+export const SET_TIMEOUT = 'SET_ SET_TIMEOUT'
```
Comment (ert485) [inline src/store/modules/slide/mutation-types.js]: copy paste error? should be ~SET_~ SET_TIMEOUT?

**Review by ert485 — CHANGES_REQUESTED** [review]

**Comment by ert485** [comment]
Does this depend on Features/connect-user-api #123?

File: src/store/modules/slide/mutation-types.js (hunk 1 above)
Comment (ridwanraji) [reply src/store/modules/slide/mutation-types.js]: FIXED! 😎

**Comment by ridwanraji** [comment]
@ert485 Not really, Its dependencies has already been added to this branch

**Comment by ert485** [comment]
>@ert485 Not really, Its dependencies has already been added to this branch

Right, but I mean there are commits in here that should be approved on another PR since they aren't part of your work :)

**Review by ert485 — APPROVED** [review]
👍

**Review by amy-kwan — APPROVED** [review]

**Review by Masonfrith — APPROVED** [review]

**Review by KingMackle — APPROVED** [review]

**Pull request merged**